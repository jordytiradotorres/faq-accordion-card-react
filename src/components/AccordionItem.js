import React, { useEffect, useRef, useState } from "react";

const AccordionItem = ({ details }) => {
  const [show, setShow] = useState(false);
  const h3 = useRef();

  const { title, description } = details;

  useEffect(() => {
    show
      ? (h3.current.style.fontWeight = "700")
      : (h3.current.style.fontWeight = "500");
  }, [show]);

  return (
    <div className="accordion__content__wrapper">
      <div className="accordion__item">
        <div className="content-show" onClick={() => setShow(!show)}>
          <h3 ref={h3}>{title}</h3>
          <button type="button" className="arrow-show">
            <img
              src={
                process.env.PUBLIC_URL + "/assets/images/icon-arrow-down.svg"
              }
              alt="icon arrow down"
            />
          </button>
        </div>

        <p className={show ? "show" : "parrafo hidden"}>{description}</p>
        <hr />
      </div>
    </div>
  );
};

export default AccordionItem;
