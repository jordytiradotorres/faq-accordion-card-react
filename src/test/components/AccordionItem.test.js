import { shallow } from "enzyme";
import AccordionItem from "../../components/AccordionItem";

describe("Pruebas en el archivo AccordionItem.js", () => {
    const title = "titulo";
    const description = "descripcion ";

    const wrapper = shallow(
        <AccordionItem title={title} description={description} />
    );

    test("Debe de renderizar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Deben de coincidir el contenido del h3 y el p", () => {
      const h3Text = wrapper.find("h3").text()
      const pText = wrapper.find(".parrafo").text()

      expect(h3Text).toBe(title)
      expect(pText).toBe(description)
  })
});
