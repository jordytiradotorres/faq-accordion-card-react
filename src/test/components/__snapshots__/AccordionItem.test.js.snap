// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo AccordionItem.js Debe de renderizar correctamente 1`] = `
<div
  className="accordion__content__wrapper"
>
  <div
    className="accordion__item"
  >
    <div
      className="content-show"
      onClick={[Function]}
    >
      <h3>
        titulo
      </h3>
      <button
        className="arrow-show"
        type="button"
      >
        <img
          alt="icon arrow down"
          src="/assets/images/icon-arrow-down.svg"
        />
      </button>
    </div>
    <p
      className="parrafo hidden"
    >
      descripcion 
    </p>
    <hr />
  </div>
</div>
`;
